sdk/assertion_test.go

```diff
@@ -33,3 +34,63 @@ func TestTDFWithAssertion(t *testing.T) {
 
 	assert.Equal(t, "4a447a13c5a32730d20bdf7feecb9ffe16649bc731914b574d80035a3927f860", string(hashOfAssertion))
 }
+
+func TestTDFWithAssertionJsonObject(t *testing.T) {
```

Comment:
nit: can we add another test where the assertion value is a string to make sure the new json.unmarshal behaves correctly in both cases

Reply:
will add a test on a later pull